src/chatdDb.h

```diff
@@ -364,6 +364,54 @@ class ChatdSqliteDb: public chatd::DbInterface
         msg.assign(buf, stmt.intCol(0), stmt.uint64Col(3), stmt.intCol(1), stmt.uint64Col(4));
     }
 
+    virtual void setOpenChat()
+    {
+        mDb.query(
+            "insert or replace into chat_vars(chatid, name, value) "
+            "values(?, 'open_chat', true)", mChat.chatId());
+        assertAffectedRowCount(1);
+    }
+
+    //Remove open_chat and preview_mode records related to the chatId
+    virtual void removeOpenChat()
+    {
+        mDb.query("delete from chat_vars where chatid = ?", mChat.chatId());
```

Comment:
The purpose of this function is to remove records for `open_chat` (and perhaps `preview_mode`) from `chat_vars` table when the chat becomes private, right? If so, you don't want to remove any other variable related to this chat, i.e. the `have_all_history`, which would still be valid for the (now) private chatroom, since we would try to keep the history in the transition from open to closed/private.